transport/tchannel/handler.go

```diff
@@ -119,13 +120,15 @@ func (h handler) handle(ctx context.Context, call inboundCall) {
 	if yarpcerrors.FromError(err).Code() == yarpcerrors.CodeResourceExhausted {
 		// all TChannel clients will time out instead of receiving an error
 		call.Response().Blackhole()
+		responseWriter.ReleaseBuffer()
```

Comment:
can we do `defer responseWriter.ReleaseBuffer()` right after the `newResponseWriter` line, and not have any of the other changes..

Reply:
Thanks @prashantv. I thought about this but did not use `defer` because in success case `responseWriter.Close()` will also put the buffer back to bufferpool. There may be unwanted side effect if we put same buffer twice to the bufferpool.

Reply:
I think the above makes sense -- we should only have a single place where we put things back in the buffer pool, so `Close` would no longer do it.

While `Close` is exported, it's part of a private type that implements a private interface, so I don't think this is actually an exported function to users. @AlexAPB can help confirm that.

@AlexAPB does this approach seem reasonable to you?

Reply:
> While Close is exported, it's part of a private type that implements a private interface, so I don't think this is actually an exported function to users.

Yup this is true. The only methods exposed on the response writer are on [this interface](https://engdocs.uberinternal.com/yarpc/pkg/go.uber.org/yarpc/api/transport/#ResponseWriter), changing functionality here is okay.

> @AlexAPB does this approach seem reasonable to you?
Yes, let's do this as it will simplify the code and ensure we don't forget a call to `ReleaseBuffer` in the future, if any of this code changes. @gopalmor can you remove all of the calls to ReleaseBuffer(), in favor of a single defer, immediately after the creation of the response writer?

Reply:
Thanks @prashantv and @AlexAPB. I have resubmitted changes as per above suggestions.